### PR TITLE
Refactor slip dropping to use throwing

### PR DESCRIPTION
**Pull request by Clyybber**
## About the PR 

Dropping hand items while slipping now reuses throwing code, which already has a LandEvent.

**Changelog**

:cl:
- tweak: Reworked how dropping items by slipping works.

### CODE REVIEW

File: Content.Server/Fluids/EntitySystems/SpillableSystem.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+public class SpillableSystem : EntitySystem
+{
+    [Dependency] private readonly SolutionContainerSystem _solutionContainerSystem = default!;
+    public override void Initialize() {
+        base.Initialize();
+        SubscribeLocalEvent<SpillableComponent, LandEvent>(SpillOnLand);
+    }
+    void SpillOnLand(EntityUid uid, SpillableComponent component, LandEvent args) {
+        if (args.User != null && Get<SolutionContainerSystem>().TryGetSolution(uid, component.SolutionName, out var solutionComponent))
+        {
```
Comment (metalgearsloth) [inline Content.Server/Fluids/EntitySystems/SpillableSystem.cs:23]: Son what is this formatting

File: Content.Server/Fluids/EntitySystems/SpillableSystem.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            Get<SolutionContainerSystem>()
```
Comment (metalgearsloth) [inline Content.Server/Fluids/EntitySystems/SpillableSystem.cs:24]: This + above just use the already cached system

File: Content.Server/Fluids/EntitySystems/SpillableSystem.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+                .Drain(uid, solutionComponent, solutionComponent.DrainAvailable)
+                .SpillAt(EntityManager.GetEntity(uid).Transform.Coordinates, "PuddleSmear");
```
Comment (metalgearsloth) [inline Content.Server/Fluids/EntitySystems/SpillableSystem.cs:26]:
```suggestion
                .SpillAt(EntityManager.GetComponent<TransformComponent>(uid).Coordinates, "PuddleSmear");
```

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.Server/Fluids/EntitySystems/SpillableSystem.cs [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+using Robust.Shared.IoC;
+
+namespace Content.Server.Fluids.EntitySystems;
+
+[UsedImplicitly]
+public class SpillableSystem : EntitySystem
+{
+    [Dependency] private readonly SolutionContainerSystem _solutionContainerSystem = default!;
+
+    public override void Initialize() {
```
Comment (ElectroJr) [inline Content.Server/Fluids/EntitySystems/SpillableSystem.cs:18]:
```suggestion
    public override void Initialize() 
    {
```

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**